akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/HttpChallenge.scala

```diff
@@ -13,7 +13,8 @@ final case class HttpChallenge(scheme: String, realm: String,
                                params: Map[String, String] = Map.empty) extends jm.headers.HttpChallenge with ValueRenderable {
```

Comment:
Bad news, that factory method isn't going to fly as is:
```def apply(scheme: String, realm: Option[String], params: Map[String, String] = Map.empty)```
>> [error] C:\Users\developer\Source\Repos\akka\akka-http-core\src\main\scala\akka\http\scaladsl\model\headers\HttpChallenge.scala:25: in obj
ect HttpChallenge, multiple overloaded alternatives of method apply define default arguments.
[error] object HttpChallenge {
[error]        ^
[error] one error found

The compiler isn't smart enough to realize the default positional arguments are the same. I guess we either drop the default Map.empty or swap parameter positions:
```def apply(scheme: String, realm: Option[String], params: Map[String, String])```
or 
```def apply(scheme: String, params: Map[String, String] = Map.empty, realm: Option[String])```
but neither looks good

Unless you have any other idea, perhaps ```realm: Option[String]``` will need to wait for another day?